boa3/compiler/compiledmetadata.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import annotations
+
+from typing import Union
+
+from boa3 import constants
+from boa3.builtin.compile_time import NeoMetadata
+from boa3.neo3.core.types import UInt160
+
+
+class CompiledMetadata:
+    _instance = None
+
+    @classmethod
+    def instance(cls) -> CompiledMetadata:
```

Comment:
I don't understand why this `classmethod` needs to exist. Wouldn't this class also work if `metadata` was a class variable instead of an instance variable?